core/sched/scheduler.c

```diff
@@ -132,7 +127,12 @@ void sched_start(void)
 
     /* Select the first guest context to switch to. */
     uint64_t expiration = 0;
-    vcpu = vcpu_find(__policy[pcpu]->do_schedule(&expiration));
+    uint32_t vcpuid = VCPUID_INVALID;
+
+    /* NOTE : Hanging on while if there's no vcpu to schedule on scheduler */
+    while( (vcpuid = __policy[pcpu]->do_schedule(&expiration)) == VCPUID_INVALID) ;
```

Comment:
It's too long and looks odd. Change it into three lines as below:
vcpuid = __policy[pcpu]->do_schedule(&expiration);
while(vcpuid == VCPUID_INVALID) 
  ;